tornado/web.py

```diff
@@ -997,7 +997,8 @@ def finish(self, chunk=None):
             if self._status_code in (204, 304):
                 assert not self._write_buffer, "Cannot send body with %s" % self._status_code
                 self._clear_headers_for_304()
-            elif "Content-Length" not in self._headers:
+            elif ("Content-Length" not in self._headers and
```

Comment:
Why did you do it differently here than in the other file? I think we should combine this with the previous `_status_code` check so that the "Cannot send body with 101" error will be generated if a body is provided.